### PR TITLE
[memgraph-2-15-1 < T516] Add text search documentation

### PR DESCRIPTION
**Pull request by antepusic**
### Description

This PR adds documentation about text indexing and search.

### Pull request type

Please check what kind of PR this is:

- [x] Fix or improvement of an existing page
- [x] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: https://github.com/memgraph/memgraph/pull/1603 https://github.com/memgraph/memgraph/pull/1739
(especially necessary if the PR is related to a release)

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [ ] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [ ] If release-related, add release note on product PR

### CODE REVIEW

File: pages/custom-query-modules/cpp/cpp-api.md [hunk 1]
```diff
@@ -2376,3 +2424,4 @@ During operation, the following exceptions may be thrown.
 | `ImmutableObjectException`    | Object you are trying to change is immutable!   |
 | `ValueConversionException`    | Error in value conversion!                      |
 | `SerializationException`      | Error in serialization!                         |
+| `TextSearchException`         | various (reports text search utility issues)    |
```
Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2427]: @antepusic Is this the actual message or do you want to say that in the case of `TextSearchException` various messages can appear?

File: pages/custom-query-modules/cpp/cpp-api.md [hunk 2]
```diff
@@ -2354,6 +2354,54 @@ Additionally, operator<< is overloaded for Type enum, and usage of this operator
 std::ostream &operator<<(std::ostream &os, const mgp::Type &type)
 ```
 
+## Database internals API
+
+This section covers the C++ API methods for interfacing with the database other
```
Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2359]: This section describes C++ API methods for database operations beyond graph manipulation.

Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2359]:
```suggestion
This section describes C++ API methods for database operations beyond graph manipulation.
```

File: pages/custom-query-modules/cpp/cpp-api.md [hunk 3]
```diff
@@ -2354,6 +2354,54 @@ Additionally, operator<< is overloaded for Type enum, and usage of this operator
...
+than with the graph itself.
```
Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2360]:
```suggestion

```

Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2360]:
```suggestion
This section describes C++ API methods for database operations beyond graph manipulation.
```

File: pages/custom-query-modules/cpp/cpp-api.md [hunk 4]
```diff
@@ -2354,6 +2354,54 @@ Additionally, operator<< is overloaded for Type enum, and usage of this operator
...
+
+### Text search
+
+#### SearchTextIndex
+
+Search the named text index for the given query and get a list of the nodes
+whose text-indexed properties match the given query.
```
Comment (kgolubic) [inline pages/custom-query-modules/cpp/cpp-api.md:2367]: Query the specified text index and retrieve a list of nodes with properties matching the query in the text index.

**Comment by kgolubic** [comment]
@antepusic I've made some changes and updates. Please take a look at everything and let me know if you are ok with the changes.

File: pages/custom-query-modules/cpp/cpp-api.md (hunk 1 above)
Comment (antepusic) [reply pages/custom-query-modules/cpp/cpp-api.md:2427]: The latter – various error messages can appear

**Review by kgolubic — APPROVED** [review]
Everything looks ok.

**Pull request merged**